### DISCUSSION TITLE
How to structure data zarr way

### DISCUSSION THREAD

**Discussion by hinxx** [zarr-developers/zarr-python#2004] [Q&A]
I have a scientific data that comes as 1 or 2 dimensional array of samples (timeseries). Along with that there is also a timestamp, one per whole array. Sometimes there would be additional information "attached" to the array represented by a scalar value(s). 

In the world of zarr how would one want to structure such data?

Should an array and scalars be made into an object? Or would they exist in groups / datasets individually?

**Comment by rabernat** [comment]
Welcome @hinxx!

Zarr doesn't specify a lot of constraints about how your structure your data.

However, this sounds like a good fit for [Xarray](https://docs.xarray.dev/en/stable/). I'd recommend modeling your data as an Xarray dataset with dimension coordinates, data variables, and attributes. Then you can use Xarray to read and write this structure into Zarr files.

**Reply by hinxx** [reply]
Thanks @rabernat , I'll take a look at xarray.
